zkdb/src/storage-engine/ipfs.ts

```diff
@@ -291,6 +291,30 @@ export class StorageEngineIPFS implements TIPFSFileSystem, TIPFSFileIndex {
     this.pathBase = path;
   }
 
+  // Try to resolve IPNS entry to CID
+  public async tryResolve() {
+    if (typeof (await this.resolve()) === 'undefined') {
+      await this.initializeMetadata();
+    }
+  }
+
+  // Initial missing data
+  public async initializeMetadata() {
+    if (!fs.existsSync(this.pathStorageMetadata)) {
+      const docMetadata: { [key: string]: string } = {};
+      const metadataContent = BSON.serialize(docMetadata);
+      fs.writeFileSync(this.pathStorageMetadata, metadataContent);
```

Comment:
We can use async alternative fs.promises, which does not block the loop

Reply:
Please create a task to solve this issue and put it in backlog. Btw, these method is in `async function` so I don't think it will block other routine.